### ISSUE TITLE
Illegal free when something goes wrong.

### ISSUE THREAD

**Issue by hdeveer** [Henneberg-Systemdesign/cp2130#6]
There is a potential issue in spi-cp2130.c @ line 776 (int cp2130_spi_transfer_one_message(...)):
```
if (chn_id == CP2130_NUM_GPIOS)
    goto err;
```
Here urb is not allocated yet and most likely causes the driver/kernel to crash when this condition arises.

It can be solved by initializing it at the top: char *urb = NULL;
and check the urb at line 835:
```
if (urb)
    kfree(urb);
```

Another issue is located in the function: int cp2130_probe(..)
The pointer *spi_master may be a bogus value if the first kzalloc() fails, the spi_master may be free-ed (after label err_out: ) because it is a random value causing a potential kernel crash too.
The solution is to initialize the variable to NULL too.

**Comment by jh-hsd** [comment]
I will take a look at the mentioned issues and review all allocs and frees the next days.

**Comment by jh-hsd** [comment]
I pushed some commits to fix those issues. Please review and let me know if the issue is resolved.

**Comment by hdeveer** [comment]
Issues are all solved in commit 72da4e3.
